pkg/cmd/experimental/syncgroups/grouplister.go

```diff
@@ -0,0 +1,123 @@
+package syncgroups
+
+import (
+	"fmt"
+
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/util"
+
+	osclient "github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/experimental/syncgroups/interfaces"
+	syncutil "github.com/openshift/origin/pkg/cmd/experimental/syncgroups/util"
+	ouserapi "github.com/openshift/origin/pkg/user/api"
+)
+
+// NewAllOpenShiftGroupLister returns a new AllLocalGroupLister
+func NewAllOpenShiftGroupLister(ldapURL string, groupClient osclient.GroupInterface) interfaces.LDAPGroupLister {
+	return &AllLocalGroupLister{
+		client:  groupClient,
+		ldapURL: ldapURL,
+	}
+}
+
+// AllLocalGroupLister lists unique identifiers for LDAP lookup of all local OpenShift Groups that
+// have been marked with an LDAP URL annotation as a result of a previous sync.
+type AllLocalGroupLister struct {
```

Comment:
Wasn't this going to be a `GroupNameMapper` too, so that you could be certain that when you resync'ed, the same openshift group was updated?